### PR TITLE
Intimate Merger Universal Identifier System: add imppid

### PR DESCRIPTION
**Pull request by eknis**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- add imppid
- update test
- refactoring

### CODE REVIEW

**Comment by eknis** [comment]
Related:
- https://github.com/prebid/Prebid.js/pull/8729

rebuilt to use existing features

**Comment by patmmccann** [comment]
@eknis if your goal is for pubs to use this as gam ppid, please feel free to demostrate that in your docs

File: test/spec/modules/imuIdSystem_spec.js [hunk 1]
```diff
@@ -122,6 +137,7 @@ describe('imuId module', function () {
     it('should return the undefined when success response', function () {
       const res = apiSuccessProcess({
         uid: 'test',
+        ppid: 'test',
```
Comment (patmmccann) [inline test/spec/modules/imuIdSystem_spec.js:140]: you might want to add a test that it can successfully get set as the gam ppid

Comment (patmmccann) [reply test/spec/modules/imuIdSystem_spec.js:140]: maybe next to this one https://github.com/prebid/Prebid.js/blob/d46661a4aa8994d57f3bea3055bd4e4c8aa50e10/test/spec/modules/userId_spec.js#L402

File: modules/userId/eids.js [hunk 2]
```diff
@@ -277,6 +277,11 @@ export const USER_IDS_CONFIG = {
     atype: 3
   },
 
+  'imppid': {
+    source: 'ppid.intimatemerger.com',
+    atype: 1
```
Comment (patmmccann) [inline modules/userId/eids.js:282]: you likely want unit tests for this too in the eids spec file

File: test/spec/modules/imuIdSystem_spec.js (hunk 1 above)
Comment (eknis) [reply test/spec/modules/imuIdSystem_spec.js]: Thanks for the review.
Added "imppid" spec.

File: modules/userId/eids.js (hunk 2 above)
Comment (eknis) [reply modules/userId/eids.js:282]: add test

File: test/spec/modules/userId_spec.js [hunk 3]
```diff
@@ -411,6 +412,7 @@ describe('User ID', function () {
             { name: 'amxId', value: {'amxId': 'amx-id-value-amx-id-value-amx-id-value'} },
             { name: 'pubCommonId', value: {'pubcid': 'pubCommon-id-value-pubCommon-id-value'} },
             { name: 'identityLink', value: {'idl_env': 'identityLink-id-value-identityLink-id-value'} },
+            { name: 'imppid', value: {'imppid': 'imppid-value-imppid-value-imppid-value'} },
```
Comment (patmmccann) [inline test/spec/modules/userId_spec.js:415]: right now this test still expects      expect(window.googletag._ppid).to.equal('pubCommonidvaluepubCommonidvalue'); on line 423; i think you want a second test where line 410 has ppid: 'yoursource.com', and an  expect(window.googletag._ppid).to.equal('yourvalue');

Comment (eknis) [reply test/spec/modules/userId_spec.js]: Added a test to make sure "imppid" is set correctly.

**Review by patmmccann — APPROVED** [review]

**Pull request merged**